framework/wazuh/core/cluster/agents_reconnect.py

```diff
@@ -30,107 +36,284 @@ def __init__(self, logger, nodes, blacklisted_nodes, workers_stability_threshold
+            self.current_phase = AgentsReconnectionPhases.NOT_ENOUGH_NODES
+
             return False
 
-        current_worker_list = set(self.nodes) - self.blacklisted_nodes
-        self.logger.debug(f"Current detected workers: {current_worker_list}.")
+        self.logger.debug(f"Current detected nodes: {node_list}.")
 
-        if self.previous_workers == current_worker_list or len(self.previous_workers) == 0:
-            if self.workers_stability_counter < self.workers_stability_threshold:
-                self.workers_stability_counter += 1
-            if self.previous_workers == set():
-                self.previous_workers = current_worker_list
+        if self.previous_nodes == node_list or len(self.previous_nodes) == 0:
+            if self.nodes_stability_counter < self.nodes_stability_threshold:
+                self.nodes_stability_counter += 1
+            if self.previous_nodes == set():
+                self.previous_nodes = node_list.copy()
         else:
-            self.logger.info("Workers changed, restarting workers stability phase.")
-            self.previous_workers = current_worker_list
+            self.logger.info("Nodes changed, restarting nodes stability phase.")
+            self.previous_nodes = node_list.copy()
             await self.reset_counter()
 
-        self.last_workers_stability_check = utils.get_utc_now()
-        if self.workers_stability_counter >= self.workers_stability_threshold:
-            self.logger.info(f"Cluster is ready {self.workers_stability_counter}/{self.workers_stability_threshold}. "
-                             f"Workers stability phase finished at "
-                             f"{self.last_workers_stability_check.strftime(DECIMALS_DATE_FORMAT)}.")
+        self.last_nodes_stability_check = utils.get_utc_now()
+        if self.nodes_stability_counter >= self.nodes_stability_threshold:
+            self.logger.info(f"Cluster is ready {self.nodes_stability_counter}/{self.nodes_stability_threshold}. "
+                             f"Nodes stability phase finished at "
+                             f"{self.last_nodes_stability_check.strftime(DECIMALS_DATE_FORMAT)}.")
             return True
 
-        self.logger.info(f"Workers are not stable at this moment. "
-                         f"Counter: {self.workers_stability_counter}/{self.workers_stability_threshold}.")
+        self.logger.info(f"Nodes are not stable at this moment. "
+                         f"Counter: {self.nodes_stability_counter}/{self.nodes_stability_threshold}.")
         return False
 
+    async def get_agents(self) -> dict:
+        """Get all agents in the system flagged as active.
+
+        Returns
+        -------
+        dict
+            Dictionary with agents information.
+        """
+        lc = local_client.LocalClient()
+        return await control.get_agents(lc, filter_status=["active"],
+                                        select_fields={'id', 'node_name', 'lastKeepAlive'})
```

Comment:
I suggest using the `forward_function` function in combination with the `get_agents` function from the agent SDK.